src/App.jsx

```diff
@@ -7,16 +7,20 @@ import {
   createBrowserRouter,
   createRoutesFromElements,
 } from "react-router-dom";
-import "./App.scss";
 import Profile from "./Pages/Profile";
-import RootLayout from "./components/RootLayout/RootLayout";
+import Header from "./components/Header/Header";
+import PrivateRoute from "./components/PrivateRoute/PrivateRoute";
+import { useSelector } from "react-redux";
 
 function App() {
+  const { authUser } = useSelector((state) => state.auth);
   const router = createBrowserRouter(
     createRoutesFromElements(
-      <Route path="/" element={<RootLayout />}>
-        <Route path="/home" element={<Home />} />
-        <Route path="/profile" element={<Profile />} />
+      <Route path="/" element={<Header />}>
+        <Route index path="/" element={<Home />} />
+        <Route path="/" element={<PrivateRoute />}>
+          <Route path={authUser.username} element={<Profile />} />
```

Comment:
What would happen when there's no authUser? `authUser.username` this part won't return undefined error then ?

Reply:
Yeah bhai, it returns an error.  Would I use statically `/me` route?

Reply:
yes @njm-wtag